utils/sparse.py

```diff
@@ -158,3 +162,56 @@ def _apply_one_shot(self):
             _LOGGER.info(f"Training recipe for one-shot application not recognized by the manager. Got recipe: "
                          f"{self.train_recipe}"
                          )
+
+    def save_sample_inputs_outputs(
+        self,
+        dataloader: "Dataloader",  # flake8 : noqa F8421
+        num_export_samples=100,
+        save_dir: Optional[str] = None,
+    ):
+
+        save_dir = save_dir or ""
+        if not dataloader:
+            raise ValueError(
+                f"Expected a data loader for exporting samples. Got {dataloader}"
+            )
+
+        device = "cuda" if torch.cuda.is_available() else "cpu"
+        exported_samples = 0
+
+        sample_in_dir = os.path.join(save_dir, "sample-inputs")
+        sample_out_dir = os.path.join(save_dir, "sample-outputs")
+
+        os.makedirs(sample_in_dir, exist_ok=True)
+        os.makedirs(sample_out_dir, exist_ok=True)
+
+        for _, (images, _, _, _) in enumerate(dataloader):
+            images = (
+                images.float() / 255
```

Comment:
To be thorough, we may want to add multi_scaling here if it was turned on for the run. See lines 176 - 182 in #40